[T12-3] MediTabs

JonathanLeeWH opened the pull request:
@jtankw3 @flxffy @melpulomas

nhs-work commented:
Feedback for v1.1:

I have gone through your project page, the readme as well as the about us section and all the required sections have been updated! Keep up the good work!
 
User Guide/Developer Guide:
Comments will be provided in separate PDF documents that will be uploaded to Luminus.

JonathanLeeWH commented:
@nhs-work On behalf of my group, Thank you for your reply.

nhs-work commented:
Feedback for v1.2:

Overall great job!

I have skimmed through the implementation section and noted some minor issues for the diagrams:

In general, the text in the diagrams is quite small making it difficult to read.

![image](https://user-images.githubusercontent.com/35650824/54880834-6d6cb080-4e84-11e9-82b1-03aced014525.png)

 
The self invocation arrows are drawn incorrectly here.

![image](https://user-images.githubusercontent.com/35650824/54880836-7198ce00-4e84-11e9-8cbe-f26c90a9ceec.png)

 
There is no indication of what is invoking the second activation bar for warning panel.

Something I noted from the code header comments of CSV wrapper that appears to be missing from the User Guide:

![image](https://user-images.githubusercontent.com/35650824/54880842-765d8200-4e84-11e9-91d9-e4f6c3f25bfe.png)
 

I do not believe there was any mention of this within the User Guide?

JonathanLeeWH commented:
@nhs-work Thank you for your feedback. With regards to the UserGuide involving CsvWrapper, I will try to update the UserGuide.adoc with that information, I did not realise I miss writing it in the User Guide though I wrote it while creating CsvWrapper.java I sincerely apologised. Thank you for pointing it out, I will be more careful in the future.

Is it possible to also help us check the Travis portion of CS2103T module repository as our own Travis Integration works fine but for some reason https://github.com/nus-cs2103-AY1819S2/addressbook-level4/ Travis integration keep showing errors and based on the logs and checking our project those files mentioned in the logs are not present in our project so we believe it is the https://github.com/nus-cs2103-AY1819S2/addressbook-level4/ repository side. I tried rerunning the Travis build for https://github.com/nus-cs2103-AY1819S2/addressbook-level4/ but it still shows the same error.

jtankw3 committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S2-T12-3/main

JonathanLeeWH committed:
Resolved Merge remote-tracking branch 'upstream/master' into update-documentation

# Resolved Conflicts:
#	docs/UserGuide.adoc

JonathanLeeWH committed:
Update UserGuide.adoc

Update UserGuide.adoc to match the new structure

JonathanLeeWH committed:
Merge branch 'master' into update-export-command

JonathanLeeWH committed:
Update UserGuide.adoc

Update Exporting as CSV file format description under Features section

flxffy committed:
UG

flxffy committed:
Merge pull request #148 from jtankw3/master

Add new sort command for BatchTable

flxffy committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S2-T12-3/main into v1.3

jtankw3 committed:
Merge pull request #153 from flxffy/v1.3

V1.3 User Guide

JonathanLeeWH committed:
Resolved any conflicts and Merge pull request #153 from flxffy/v1.3

JonathanLeeWH committed:
Revert "Resolved any conflicts and Merge pull request #153 from flxffy/v1.3"

This reverts commit 5bd5aff05a80ec183a5c78c8b61672c2c694aec7.

JonathanLeeWH committed:
Merge branch 'master' into update-documentation

jtankw3 committed:
Add draft 1 of PPP

jtankw3 committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S2-T12-3/main

JonathanLeeWH committed:
Update UserGuide.adoc and DeveloperGuide.adoc

Update stylesheet to fix printing in PDF
Update UserGuide.adoc file naming convention description
Update DeveloeprGuide.adoc file naming convention description
Update AboutUs.adoc
Upload my Project Portfolio

JonathanLeeWH committed:
Update formatting

jtankw3 committed:
Fix some small bugs

jtankw3 committed:
Merge pull request #150 from JonathanLeeWH/update-documentation

Update documentation

JonathanLeeWH committed:
Merge branch 'master' into update-export-command

jtankw3 committed:
Update UserGuide for v1.3

jtankw3 committed:
Update PPP

jtankw3 committed:
Fix typo in user guide

jtankw3 committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S2-T12-3/main

jtankw3 committed:
Remove Placeholder from UserGuide

jtankw3 committed:
Rename PPP file

JonathanLeeWH committed:
Update AboutUs.adoc

Update team members Project Portfolio link

JonathanLeeWH committed:
Merge pull request #154 from jtankw3/master

Documentation for V1.3

JonathanLeeWH committed:
Merge branch 'master' into project-portfolio

JonathanLeeWH committed:
Delete dummy file

JonathanLeeWH committed:
Merge branch 'project-portfolio' of https://github.com/JonathanLeeWH/main into project-portfolio

JonathanLeeWH committed:
Merge pull request #155 from JonathanLeeWH/project-portfolio

Update AboutUs.adoc

melpulomas committed:
Added adoc

melpulomas committed:
Spelling error

jtankw3 committed:
Merge branch 'master' into master

flxffy committed:
ppp first draft

JonathanLeeWH committed:
Merge branch 'master' into v1.3

JonathanLeeWH committed:
Merge pull request #156 from melpulomas/master

Added adoc

flxffy committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S2-T12-3/main into v1.3

flxffy committed:
typo

JonathanLeeWH committed:
Merge pull request #157 from flxffy/v1.3

PPP first draft

JonathanLeeWH committed:
Merge branch 'master' into update-export-command

JonathanLeeWH committed:
Update jonathanleewh.adoc

Update jonathanleewh.adoc (my project portfolio) to ensure consistency, formatting, pharsing and language

JonathanLeeWH committed:
Update ExportCommandTest.java

Update ExportCommandTest.java compareActualAndExpectedData() test method with additional comments on why I did not take advantage of JUnit's API TemporaryFolder (likely to be used beyond milestone v2.0)

JonathanLeeWH committed:
Merge pull request #158 from JonathanLeeWH/update-export-command

Update Junit test cases for export command

flxffy committed:
fixed warning command parser test

JonathanLeeWH committed:
Update jonathanleewh.adoc

Update jonathanleewh.adoc based on CS2101 module tutor's comments

JonathanLeeWH committed:
Update DeveloperGuide.adoc

Change filter command to find command in export description (design considerations)

JonathanLeeWH committed:
Update jonathanleewh.adoc

Update jonathanleewh.adoc to ensure consistency

JonathanLeeWH committed:
Update UserGuide.adoc

Update UserGuide.adoc integration addon support description

JonathanLeeWH committed:
Update UserGuide.adoc

Update UserGuide.adoc integration addon support description (Benefits) and include figure caption for sample of exported CSV file in the exporting as CSV file format section

JonathanLeeWH committed:
Update UserGuide.adoc

Update UserGuide.adoc exporting as CSV file format description (Information to take note and important information)

JonathanLeeWH committed:
Update UserGuide.adoc

Update UserGuide.adoc exporting as CSV file format section (important information)

flxffy committed:
date input validation

JonathanLeeWH committed:
Update UserGuide.adoc

Update UserGuide.adoc integration addon support description (Note)

flxffy committed:
fixed window size

JonathanLeeWH committed:
Update DeveloperGuide.adoc

Update DeveloperGuide.adoc exporting as CSV file format description

JonathanLeeWH committed:
Update UserGuide.adoc and DeveloperGuide.adoc

Update UserGuide.adoc and DeveloperGuide.adoc to ensure consistency

jtankw3 committed:
Add SortCommandParserTest and fix SortCommandParser bugs

jtankw3 committed:
Add SortCommandTest and fix SortCommand bugs

jtankw3 committed:
Add description for behavior if there is more than one valid input in user guide

jtankw3 committed:
Fix typo in Find command in user guide

jtankw3 committed:
Add InformationPanelSettingsTest

jtankw3 committed:
Include sorting test in InformationPanelTest

jtankw3 committed:
Fixed expiry validation so that leap year is accounted for

jtankw3 committed:
Fix update command bug that caused information panel to not change

flxffy committed:
Updated UG for clarity

flxffy committed:
set minimum resolution

jtankw3 committed:
Improve updateCommandTest to test for selectedMedicine

jtankw3 committed:
Remove trailing whitespace

jtankw3 committed:
Remove trailing white space from userguide

jtankw3 committed:
Merge branch 'master' into master

JonathanLeeWH committed:
Merge pull request #178 from jtankw3/master

Fix bugs from dry run testing session

JonathanLeeWH committed:
Merge remote-tracking branch 'upstream/master' into portfolio

flxffy committed:
sorted all lists alphabetically + fixed most tests

flxffy committed:
resolved merge conflict

flxffy committed:
existing tests partially passing

flxffy committed:
fixed style

flxffy committed:
fixed style

flxffy committed:
fixed style

jtankw3 committed:
Merge pull request #179 from flxffy/v1.4

V1.4 Sorted medicine list

jtankw3 committed:
Fix small bug in sorted medicine list

jtankw3 committed:
Fix codestyle

jtankw3 committed:
Improve FindCommand and FindCommandParser to look at other fields

jtankw3 committed:
Add Predicate package and change import statements

jtankw3 committed:
Small improvement to ExpiryTest and InformationPanelTest

jtankw3 committed:
Fix codestyle

JonathanLeeWH committed:
Merge branch 'master' into portfolio

jtankw3 committed:
Fix codestyle

jtankw3 committed:
Fix codestyle

jtankw3 committed:
Merge branch 'master' of https://github.com/jtankw3/main

jtankw3 committed:
Improve FindCommandParserTest

jtankw3 committed:
Update FindCommandTest to test for new functionality

jtankw3 committed:
Update FindCommandSystemTest for new functionality in FindCommand

jtankw3 committed:
Update FindCommand in UserGuide

jtankw3 committed:
Add InformationPanelSettings to UserPref

jtankw3 committed:
Clean up codestyle for BatchTable and InformationPanel

jtankw3 committed:
Improve InformationPanel section of developers guide

jtankw3 committed:
Fix codestyle

JonathanLeeWH committed:
Merge pull request #180 from jtankw3/master

FindCommand added

JonathanLeeWH committed:
Merge branch 'master' into portfolio

melpulomas committed:
Updated Developer Guide

melpulomas committed:
Updated user guide

melpulomas committed:
Updated Adoc

melpulomas committed:
Added box to the output medicine label

JonathanLeeWH committed:
Merge pull request #181 from JonathanLeeWH/portfolio

Update documentation

melpulomas committed:
added images for user and developer guide

melpulomas committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S2-T12-3/main

# Conflicts:
#	docs/UserGuide.adoc

melpulomas committed:
Fix import statement that cause checkstyle error

melpulomas committed:
Fixed conflict error

jtankw3 committed:
Merge pull request #182 from melpulomas/master

Mid 1.4

jtankw3 committed:
Fix bug in sort command regarding quantity

jtankw3 committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S2-T12-3/main

jtankw3 committed:
Update test with more robust test data

JonathanLeeWH committed:
Merge pull request #183 from jtankw3/master

Fix bug and improve test

JonathanLeeWH committed:
Update DeveloperGuide.adoc based on review

Update DeveloperGuide.adoc based on review
Fix link in Introduction section

JonathanLeeWH committed:
Update DeveloperGuide.adoc based on review

Update DeveloperGuide.adoc based on review
Remove {more test cases ...} from deleting a medicine section as suggested in the review
Remove the product survey section as we did not use it
Update wording for consistency `MacOS` to `macOS` as the official name is `macOS` with the letter `m` being a small letter
Update Glossary section to ensure consistency

JonathanLeeWH committed:
Update jonathanleewh.adoc

Update jonathanleewh.adoc number of managed releases

JonathanLeeWH committed:
Update UserGuide.adoc, DeveloperGuide.adoc and jonathanleewh.adoc

Update UserGuide.adoc, DeveloperGuide.adoc and jonathanleewh.adoc to ensure consistency

JonathanLeeWH committed:
Update UserGuide.adoc

Update UserGuide.adoc to ensure consistency (change `MacOS` to `macOS` as `macOS` is the official name)

JonathanLeeWH committed:
Merge pull request #185 from JonathanLeeWH/update-developer-guide

Update DeveloperGuide.adoc based on review

JonathanLeeWH committed:
Update jonathanleewh.adoc to ensure consistency

Update jonathanleewh.adoc to ensure consistency

JonathanLeeWH committed:
Update jonathanleewh.adoc

Update jonathanleewh.adoc

JonathanLeeWH committed:
Update jonathanleewh.adoc

Update jonathanleewh.adoc

JonathanLeeWH committed:
Update jonathanleewh.adoc

Update jonathanleewh.adoc

JonathanLeeWH committed:
Prepare MediTabs for v1.4 milestone

Update MediTabs version number to `1.4.0` to prepare for v1.4 milestone

jtankw3 committed:
Merge pull request #186 from JonathanLeeWH/update-documentation

Update documentation and prepare MediTabs for v1.4 milestone

JonathanLeeWH committed:
Update UserGuide.adoc

Update UserGuide.adoc with correct name for Singapore's Health Sciences Authority (HSA)

flxffy committed:
Merge pull request #187 from JonathanLeeWH/update-documentation

Update UserGuide.adoc

flxffy committed:
warning card test added

flxffy committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S2-T12-3/main into v1.4

flxffy committed:
added max expiry date

flxffy committed:
updated tests related to expiry dates

flxffy committed:
added warning list view tests

flxffy committed:
removed unused imports

flxffy committed:
fixed style

flxffy committed:
changed import order

jtankw3 committed:
Merge pull request #188 from flxffy/v1.4

V1.4 Ui test, Validation for expiry dates

flxffy committed:
changed condition for date and threshold

flxffy committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S2-T12-3/main into v1.4

flxffy committed:
catch exception for invalid threshold when loading from user pref

flxffy committed:
lambda style

flxffy committed:
lambda style

flxffy committed:
changed configuration for user pref and added test

jtankw3 committed:
Add limits to string length, number of batches and number of tags allowed

JonathanLeeWH committed:
Merge pull request #189 from flxffy/v1.4

V1.4 Changed date and threshold validation

jtankw3 committed:
Edit description for add, edit, find command

jtankw3 committed:
Make font size dynamic in informmation panel

jtankw3 committed:
Add final tests for update command and information panel

jtankw3 committed:
Merge branch 'master' into master

jtankw3 committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S2-T12-3/main

jtankw3 committed:
Move methods for testing to InformationPanelHandle

jtankw3 committed:
Change arguments for SortCommand

flxffy committed:
added error for warning commands with non whitespace preambles

melpulomas committed:
Editied developer guide for v1.4

melpulomas committed:
Edit User Guide

melpulomas committed:
Edited Label command, now users cannot hvae input of length above a limit

melpulomas committed:
edited Label Test for 1.4

melpulomas committed:
Edited User guide to include coming in v2.0

melpulomas committed:
Merge branch 'master' into master

jtankw3 committed:
Various code style changes

jtankw3 committed:
Add UpdateCommandSystemTest

jtankw3 committed:
Merge branch 'master' of https://github.com/jtankw3/main

jtankw3 committed:
Remove trailing whitespace

flxffy committed:
Merge pull request #190 from jtankw3/master

Codestyle and debugging

flxffy committed:
change uml diagram

flxffy committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S2-T12-3/main into v1.4

JonathanLeeWH committed:
Update CsvWrapper.java

Update CsvWrapper.java to include a more concise exception message when users try to input file name that is too long such that it is not within their current operating system's filesystem capability

JonathanLeeWH committed:
Update FileName.java

Update FileName.java which includes the help command in the exception message to notify users they can use the command to view the user guide

JonathanLeeWH committed:
Update UserGuide.adoc

Update UserGuide.adoc export command description

JonathanLeeWH committed:
Merge pull request #192 from JonathanLeeWH/improve-csv-wrapper

Improve Export Command

jtankw3 committed:
Change order of medicine card fields

jtankw3 committed:
Change toLowerCase and equals to equalsIgnoreCase

jtankw3 committed:
Merge branch 'master' into master

JonathanLeeWH committed:
Merge branch 'master' into master

JonathanLeeWH committed:
Merge pull request #193 from jtankw3/master

Change order of medicine card fields

flxffy committed:
update DG

jtankw3 committed:
Edit user guide introduction

jtankw3 committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S2-T12-3/main

flxffy committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S2-T12-3/main into v1.4

flxffy committed:
adjust panel sizes to view entire error message

flxffy committed:
activity diagram for warning command

jtankw3 committed:
Final changes to UserGuide for v1.4

jtankw3 committed:
Fix grammatical error in UserGuide_SortCommand.png

flxffy committed:
PPP

flxffy committed:
import order

flxffy committed:
import order

melpulomas committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S2-T12-3/main

melpulomas committed:
Shifted documentation for 1.4

melpulomas committed:
Change to constant max file length

melpulomas committed:
Merge branch 'master' of https://github.com/melpulomas/main

jtankw3 committed:
Fix UpdateCommand bug

jtankw3 committed:
Merge pull request #191 from melpulomas/master

Edited user, developer and Label command [1.4]

jtankw3 committed:
Merge branch 'master' into v1.4

jtankw3 committed:
Final updates to developers guide for v1.4

jtankw3 committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S2-T12-3/main

JonathanLeeWH committed:
Update CsvWrapper.java

Update CsvWrapper.java with a check for maximum specified file name length based on the user's current operating system or it will default to 255 character limit for file name which is the general convention by programmers as there is no universal method and is still debatable by many programmers in a portable file name length

JonathanLeeWH committed:
Update CsvWrapper.java

Update CsvWrapper.java to ensure it matches coding standards

JonathanLeeWH committed:
Update UserGuide.adoc

Update UserGuide.adoc export command description

JonathanLeeWH committed:
Update DeveloperGuide.adoc

Update DeveloperGuide.adoc with sample test cases and expected outputs for the various test cases for the export command under Instructions for Manual Testing section

JonathanLeeWH committed:
Update DeveloperGuide.adoc

Update DeveloperGuide.adoc Instructions for manual testing section export command description

JonathanLeeWH committed:
Update CsvWrapper.java

Update CsvWrapper.java description

jtankw3 committed:
Merge pull request #197 from JonathanLeeWH/improve-documentation

Improve documentation

flxffy committed:
test cases to dg

flxffy committed:
resolve conflict

flxffy committed:
Merge branch 'v1.4' of https://github.com/flxffy/main into v1.4

JonathanLeeWH committed:
Merge pull request #195 from flxffy/v1.4

V1.4 Documentation

melpulomas committed:
Fixed screenshot image

melpulomas committed:
Fix images

flxffy committed:
update ug contributions

flxffy committed:
typo

flxffy committed:
typo

melpulomas committed:
edited documentation for v1.4

flxffy committed:
Merge branch 'master' into v1.4

melpulomas committed:
Include test in adoc

melpulomas committed:
Edited adoc

melpulomas committed:
fix checkstyle

JonathanLeeWH committed:
Merge pull request #198 from flxffy/v1.4

update ug contributions

jtankw3 committed:
Update developer guide for v1.4

jtankw3 committed:
Update user guide for v1.4

jtankw3 committed:
Remove image for SortCommand

jtankw3 committed:
Fix bug caused by clicking column in sort commnad

JonathanLeeWH committed:
Merge branch 'master' into master

jtankw3 committed:
Update PPP for v1.4

jtankw3 committed:
Codestyle change in BatchTable

melpulomas committed:
Fix developer guide

melpulomas committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S2-T12-3/main

# Conflicts:
#	docs/DeveloperGuide.adoc

jtankw3 committed:
Merge branch 'master' into master

melpulomas committed:
Merge branch 'master' of https://github.com/melpulomas/main

# Conflicts:
#	docs/DeveloperGuide.adoc

melpulomas committed:
Change to png

melpulomas committed:
Fix image

JonathanLeeWH committed:
Updated DeveloperGuide.adoc

jtankw3 committed:
Add test case extract to PPP

jtankw3 committed:
Merge branch 'master' of https://github.com/jtankw3/main

melpulomas committed:
Fix user guide

jtankw3 committed:
Remove trailing space

JonathanLeeWH committed:
Updated UserGuide.adoc

Update UserGuide.adoc based on Ai hui's request credits goes to Ai hui

JonathanLeeWH committed:
Updated Ai Hui's Project Portfolio on her behalf

With permission from Ai Hui due to her laptop running out of battery, I help her edit on her behalf credits goes to her

jtankw3 committed:
remove trailing whitespace

JonathanLeeWH committed:
Update jonathanleewh.adoc

Update jonathanleewh.adoc to include including export command test in Developer Guide's Instructions for Manual Testing section

jtankw3 committed:
Update Ui.png

JonathanLeeWH committed:
Merge pull request #200 from JonathanLeeWH/master

Update documentation

JonathanLeeWH committed:
Merge branch 'master' into master

jtankw3 committed:
Merge branch 'master' into master

JonathanLeeWH committed:
Merge pull request #199 from melpulomas/master

Fix error and edit documentation for v1.4

jtankw3 committed:
Merge branch 'jtankw3-master'

jtankw3 committed:
Resolved Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	docs/UserGuide.adoc
#	docs/images/Ui.png

JonathanLeeWH committed:
Merge pull request #196 from jtankw3/master

User and Developer Guide for V1.4

jtankw3 committed:
Fix formatting issues with documentation

jtankw3 committed:
Merge branch 'master' into master

JonathanLeeWH committed:
Merge pull request #201 from jtankw3/master

Fix formatting issues with documentation
